src/com_tjvendors/site/models/vendor.php

```diff
@@ -607,4 +612,76 @@ public static function getPayableAmount($vendorId, $client = '', $currency = '')
 
 		return $payableAmount;
 	}
+
+/**
+	 * Get the vendor id
+	 *
+	 * @param   Int     $userId  user id
+	 *
+	 * @param   String  $client  client ex->"com_tjlms/com_jticketing"
+	 *
+	 * @return  Int
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function validateVendor($userId, $client)
```

Comment:
@amit4106udale the function name needs to be updated as per the functionality of the method.

The function seems to be checking if a user is already a vendor and if he is not then adding him as a vendor.

What exactly we are validating in this function?

Reply:
@ankush-maherwal - We are checking the user is a vendor or not. If the user is not a vendor then make him. if he is a vendor the returning the id as it is. Discussed this with @twsvaishali @thite-amol

Reply:
@amit4106udale Seems the function is wrongly named....validateVendor sounds as if we are performing some kind of validation for the vendor (Creating the vendor should not be the part of this function). To check if a user is a vendor for some client we already have a function named "checkVendor"